### PR TITLE
update mypy version

### LINKED ISSUE #243: Update to mypy 1
https://github.com/PolicyEngine/policyengine-core/blob/47f5bdf953d8c6776f9af1f3859864c12ebbf36f/setup.py#L38

Currently 1.11.1: https://pypi.org/project/mypy/

### PR DESCRIPTION
**Pull request by SeanAverS**
## What's changed

in `setup.py`, update mypy version to `<2`

## What this fixes and how it's fixed
- Fixes #243

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -35,7 +35,7 @@
     "markupsafe==2.0.1",
     "coverage",
     "furo",
-    "mypy==0.991",
+    "mypy==1.11.1",
```
Comment (MaxGhenis) [inline setup.py:38]: We generally only pin to a specific version if an incompatibility arises, otherwise the current major
```suggestion
    "mypy<2",
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Thanks! Did this pass tests locally?

**Comment by MaxGhenis** [comment]
Please also populate changelog_entry.yaml (see other PRs)

**Comment by MaxGhenis** [comment]
changelog_entry.yaml not changelog.yaml

**Comment by SeanAverS** [comment]
@MaxGhenis thank you for your patience 🙏

**Comment by MaxGhenis** [comment]
Please do not edit changelog (we automate that)

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
please match the format of changelog_entry.yaml from other merged prs to address the CI failure

File: changelog_entry.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
```
Comment (MaxGhenis) [inline changelog_entry.yaml:1]:
```suggestion
- bump: patch
```
Since it doesn't introduce new functionality

File: changelog_entry.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    changed:
+    - change mypy version from 1.11.1 to <2
```
Comment (MaxGhenis) [inline changelog_entry.yaml:4]:
```suggestion
      - change mypy version from 1.11.1 to <2

```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**